pkg/api/meta/meta.go

```diff
@@ -480,6 +485,52 @@ func extractFromTypeMeta(v reflect.Value, a *genericAccessor) error {
 	return nil
 }
 
+func setField(src reflect.Value, fieldName string, dest interface{}) error {
+	field := src.FieldByName(fieldName)
+	if !field.IsValid() {
+		return fmt.Errorf("%v does not have field %s", src, fieldName)
+	}
+	v, err := conversion.EnforcePtr(dest)
+	if err != nil {
+		return err
+	}
+	if field.Type().AssignableTo(v.Type()) {
+		v.Set(field)
+		return nil
+	}
+	if field.Type().ConvertibleTo(v.Type()) {
+		v.Set(field.Convert(v.Type()))
+		return nil
+	}
+	return fmt.Errorf("couldn't assign/convert %v to %v", field.Type(), v.Type())
+}
+
+func extractUnversionedOwnerReference(v reflect.Value) (unversioned.ObjectReference, error) {
```

Comment:
@lavalamp I'm using reflection to copy whatever version of ObjectReference to unversioned.ObjectReference.